acnportal/acnsim/tests/test_interface.py

```diff
@@ -272,9 +273,8 @@ def setUp(self) -> None:
         self.allowable_rates: List[int] = [0, 8, 16, 24, 32]
         evse4: FiniteRatesEVSE = FiniteRatesEVSE("PS-004", self.allowable_rates)
         self.network.register_evse(evse4, 120, -30)
-        self.network.constraint_matrix = np.eye(4)
-        self.network.magnitudes = np.ones((4, 1))
-        self.network.constraint_index = ["C1", "C2", "C3", "C4"]
+        for i, station_id in enumerate(self.network.station_ids):
+            self.network.add_constraint(Current(station_id), 1, f"C{i+1}")
```

Comment:
On my end, add_constraint seems to break in pandas 2 due to DataFrames not having an append method.